src/hotspot/cpu/aarch64/sharedRuntime_aarch64.cpp

```diff
@@ -1038,9 +1038,50 @@ static void continuation_enter_cleanup(MacroAssembler* masm) {
   __ stop("incorrect sp1");
   __ bind(OK);
 #endif
-
   __ ldr(rscratch1, Address(sp, ContinuationEntry::parent_cont_fastpath_offset()));
   __ str(rscratch1, Address(rthread, JavaThread::cont_fastpath_offset()));
+
+  if (CheckJNICalls) {
+    // Check if this is a virtual thread continuation
+    Label L_skip_vthread_code;
+    __ ldrw(rscratch1, Address(sp, ContinuationEntry::flags_offset()));
+    __ cbzw(rscratch1, L_skip_vthread_code);
+
+    Label L_no_warn;
+    __ ldr(rscratch1, Address(rthread, JavaThread::jni_monitor_count_offset()));
+    __ cbz(rscratch1, L_no_warn);
```

Comment:
```suggestion
    __ cbz(rscratch1, L_skip_vthread_code);
```

Reply:
Same for other ports.  If the count is already 0, no need to store 0 below.

Reply:
> If the count is already 0, no need to store 0 below.

Good observation. I'm updating x64 and arrach64 and will re-test then do the other ports. Thanks